internal/auth/ldap/repository_auth_method_read.go

```diff
@@ -32,19 +34,36 @@ func (r *Repository) LookupAuthMethod(ctx context.Context, publicId string, opt
 	return r.lookupAuthMethod(ctx, publicId, WithUnauthenticatedUser(ctx, opts.withUnauthenticatedUser))
 }
 
-// ListAuthMethods returns a slice of AuthMethods for the scopeId. The
-// WithUnauthenticatedUser, WithLimit and WithOrder options are supported and
-// all other options are ignored.
-func (r *Repository) ListAuthMethods(ctx context.Context, scopeIds []string, opt ...Option) ([]*AuthMethod, error) {
+// ListAuthMethods returns a slice of AuthMethods for the scopeId. Supported options:
+//   - auth.WithLimit
+//   - auth.WithStartPageAfterItem
+func (r *Repository) ListAuthMethods(ctx context.Context, scopeIds []string, opt ...auth.Option) ([]auth.AuthMethod, error) {
 	const op = "ldap.(Repository).ListAuthMethods"
 	if len(scopeIds) == 0 {
 		return nil, errors.New(ctx, errors.InvalidParameter, op, "missing scope IDs")
 	}
-	authMethods, err := r.getAuthMethods(ctx, "", scopeIds, opt...)
+	opts, err := auth.GetOpts(opt...)
+	if err != nil {
+		return nil, err
+	}
+	newOpts := []Option{
+		WithUnauthenticatedUser(ctx, opts.WithUnauthenticatedUser),
+	}
+	if opts.WithLimit != 0 {
+		newOpts = append(newOpts, WithLimit(ctx, opts.WithLimit))
+	}
+	if opts.WithStartPageAfterItem != nil {
+		newOpts = append(newOpts, WithStartPageAfterItem(ctx, opts.WithStartPageAfterItem.PublicId, opts.WithStartPageAfterItem.UpdateTime))
+	}
```

Comment:
We have to convert the auth options to the repository specific options here before calling `getAuthMethod`, since that function is shared with `LookupAuthMethod`.